packages/@react-spectrum/s2/src/Badge.tsx

```diff
@@ -36,13 +37,15 @@ export interface BadgeStyleProps {
   variant: 'accent' | 'informative' | 'neutral' | 'positive' | 'notice' | 'negative' | 'gray' | 'red' | 'orange' | 'yellow' | 'charteuse' | 'celery' | 'green' | 'seafoam' | 'cyan' | 'blue' | 'indigo' | 'purple' | 'fuchsia' | 'magenta' | 'pink' | 'turquoise' | 'brown' | 'cinnamon' | 'silver'
 }
 
-export interface BadgeProps extends DOMProps, AriaLabelingProps, StyleProps, BadgeStyleProps{
+export interface BadgeProps extends DOMProps, AriaLabelingProps, StyleProps, BadgeStyleProps, SlotProps {
   /**
    * The content to display in the badge.
    */
   children: ReactNode
 }
 
+export const BadgeContext = createContext<ContextValue<Partial<BadgeProps>, DOMRefValue<HTMLDivElement>>>(null);
```

Comment:
why is this one Partial but so many of the others are not? seems like a useful thing to put there for all of them

Reply:
because this one has required children but the others don't